common/lib/xmodule/xmodule/js/src/html/edit.js

```diff
@@ -171,7 +172,37 @@
            */
           init_instance_callback: this.initInstanceCallback,
           browser_spellcheck: true
-        });
+        };
+
+        if (typeof process != "undefined" && process.env.JS_ENV_EXTRA_CONFIG) {
+          var tinyMceAdditionalPlugins = JSON.parse(process.env.JS_ENV_EXTRA_CONFIG).TINYMCE_ADDITIONAL_PLUGINS;
+          // check if we have any additional plugins passed
+          if (tinyMceAdditionalPlugins) {
+            // go over each plugin
+            for (var plugin_name in tinyMceAdditionalPlugins) {
+              // check if plugins is not empty (ie there are existing plugins)
+              if (tinyMceConfig.plugins.trim()) {
+                tinyMceConfig.plugins += ', ';
+              }
+
+              // add the plugin to the list of plugins
+              tinyMceConfig.plugins += plugin_name;
+
+              // check if toolbar is not empty (ie there are already items in the toolbar)
+              if (tinyMceConfig.toolbar.trim()) {
+                tinyMceConfig.toolbar += ' | ';
+              }
+              tinyMceConfig.toolbar += plugin_name;
```

Comment:
It has come to my attention that not all TinyMCE plugins will need to be present in the toolbar. I'll need to address that in a slight change.

Reply:
Alright, this has been resolved :+1: